ngmix/tests/test_em.py

```diff
@@ -0,0 +1,326 @@
+
+        irr_1 = T_1 / 2.0 * (1 - e1_1)
+        irc_1 = T_1 / 2.0 * e2_1
+        icc_1 = T_1 / 2.0 * (1 + e1_1)
+
+        irr_2 = T_2 / 2.0 * (1 - e1_2)
+        irc_2 = T_2 / 2.0 * e2_2
+        icc_2 = T_2 / 2.0 * (1 + e1_2)
+
+        pars = [
+            counts_1,
+            cen1[0],
+            cen1[1],
+            irr_1,
+            irc_1,
+            icc_1,
+            counts_2,
+            cen2[0],
+            cen2[1],
+            irr_2,
+            irc_2,
+            icc_2,
+        ]
+
+    elif ngauss == 1:
+
+        pars = [
+            counts,
+            cen1[0],
+            cen1[1],
+            irr_1,
+            irc_1,
+            icc_1,
+        ]
+
+    gm = GMix(pars=pars)
+
+    if withpsf:
+        gmconv = gm.convolve(psf_gm)
+        psf_im = psf_gm.make_image(dims, jacobian=jacob)
+        psf_obs = Observation(psf_im, jacobian=jacob)
+        psf_obs.set_gmix(psf_gm)
+
+        im0 = gmconv.make_image(dims, jacobian=jacob)
+    else:
+        im0 = gm.make_image(dims, jacobian=jacob)
+        psf_obs = None
+
+    im = im0 + rng.normal(size=im0.shape, scale=noise)
+    obs = Observation(im, jacobian=jacob, psf=psf_obs)
+
+    return obs, gm
+
+
+def get_psf_obs():
+    pixel_scale = 0.263
+    dims = [25, 25]
+    cen = (np.array(dims) - 1.0) / 2.0
+
+    jacob = DiagonalJacobian(scale=pixel_scale, row=cen[0], col=cen[1])
+
+    Tpsf = 0.27
+    psf_gm = GMixModel([0.0, 0.0, 0.0, 0.0, Tpsf, 1.0], "turb")
+    # psf_gm = GMixModel([0.0, 0.0, 0.0, 0.0, Tpsf, 1.0], "gauss")
+    psf_im = psf_gm.make_image(dims, jacobian=jacob)
+
+    return Observation(psf_im, jacobian=jacob), psf_gm
+
+
+@pytest.mark.parametrize('noise', [0.0, 0.05])
+def test_1gauss(noise):
+    """
+    see if we can recover the input with and without noise to high precision
+    even with a bad guess
+
+    Use ngmix to make the image to make sure there are
+    no pixelization effects
+    """
+
+    pixel_scale = 0.263
+    rng = np.random.RandomState(42587)
+    ngauss = 1
+    obs, gm = get_obs(rng=rng, ngauss=ngauss, pixel_scale=0.263, noise=noise)
+
+    pars = gm.get_full_pars()
+    counts = pars[0]
+
+    gm_guess = gm.copy()
+    gm_guess._data["p"] += counts/10 * srandu(rng=rng)
+    gm_guess._data["row"] += 4 * pixel_scale * srandu(rng=rng)
+    gm_guess._data["col"] += 4 * pixel_scale * srandu(rng=rng)
+    gm_guess._data["irr"] += 0.5 * pixel_scale**2 * srandu(rng=rng)
+    gm_guess._data["irc"] += 0.5 * pixel_scale**2 * srandu(rng=rng)
+    gm_guess._data["icc"] += 0.5 * pixel_scale**2 * srandu(rng=rng)
+
+    fitter = fit_em(obs=obs, guess=gm_guess)
+
+    fit_gm = fitter.get_gmix()
+    res = fitter.get_result()
+    assert res['flags'] == 0
+
+    fitpars = fit_gm.get_full_pars()
+
+    if noise == 0.0:
+        tol = 1.0e-4
+        assert (fitpars[0]/pars[0]-1) < tol
+        assert (fitpars[3]/pars[3]-1) < tol
+        assert (fitpars[4]/pars[4]-1) < tol
+        assert (fitpars[5]/pars[5]-1) < tol
```

Comment:
What about the other parameters?

Reply:
I didn't have a guess at how far we expect those to drift.

Reply:
We can use something loose, maybe 1/10 pixel

I think the only way the center could be off but the  moments be OK is if there was a bug that changed the definition of the center